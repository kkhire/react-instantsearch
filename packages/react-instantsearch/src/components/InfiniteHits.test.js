import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Connect, { InfiniteHits } from './InfiniteHits';

Enzyme.configure({ adapter: new Adapter() });

describe('InfiniteHits - Raw', () => {
  const defaultProps = {
    hits: [{ objectID: 0 }, { objectID: 1 }, { objectID: 2 }],
    hasMore: false,
    cx: (...x) => x.join(' '),
    refine: () => {},
    translate: x => x,
  };

  it('renders a default hit', () => {
    const props = {
      ...defaultProps,
    };

    const wrapper = shallow(<InfiniteHits {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders a custom hit', () => {
    const props = {
      ...defaultProps,
      renderHit: hit => <div id={hit.objectID} />,
    };

    const wrapper = shallow(<InfiniteHits {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('button is disabled when it is the last page', () => {
    const props = {
      ...defaultProps,
      hasMore: false,
    };

    const wrapper = shallow(<InfiniteHits {...props} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('button').prop('disabled')).toBe(true);
  });

  it('calls refine onClick', () => {
    const props = {
      ...defaultProps,
      refine: jest.fn(),
    };

    shallow(<InfiniteHits {...props} />)
      .find('button')
      .simulate('click');

    expect(props.refine).toHaveBeenCalled();
  });
});

describe('InfiniteHits - Connect', () => {
  const defaultProps = {
    hits: [{ objectID: 0 }, { objectID: 1 }, { objectID: 2 }],
    hasMore: false,
    cx: (...x) => x.join(' '),
    refine: () => {},
  };

  it('renders with translate', () => {
    const props = {
      ...defaultProps,
    };

    const wrapper = shallow(<Connect {...props} />).dive();

    expect(wrapper).toMatchSnapshot();
  });
});
