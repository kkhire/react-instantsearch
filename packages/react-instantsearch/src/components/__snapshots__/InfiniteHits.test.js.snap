// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfiniteHits - Connect renders with translate 1`] = `
<div>
  <ul
    className="list"
  >
    <li
      className="item"
      key="0"
    >
      <div
        style={
          Object {
            "borderBottom": "1px solid #bbb",
            "marginBottom": "5px",
            "paddingBottom": "5px",
            "wordBreak": "break-all",
          }
        }
      >
        {"objectID":0}
        ...
      </div>
    </li>
    <li
      className="item"
      key="1"
    >
      <div
        style={
          Object {
            "borderBottom": "1px solid #bbb",
            "marginBottom": "5px",
            "paddingBottom": "5px",
            "wordBreak": "break-all",
          }
        }
      >
        {"objectID":1}
        ...
      </div>
    </li>
    <li
      className="item"
      key="2"
    >
      <div
        style={
          Object {
            "borderBottom": "1px solid #bbb",
            "marginBottom": "5px",
            "paddingBottom": "5px",
            "wordBreak": "break-all",
          }
        }
      >
        {"objectID":2}
        ...
      </div>
    </li>
  </ul>
  <button
    className="loadMore loadMore--disabled"
    disabled={true}
    onClick={[Function]}
  >
    Load more
  </button>
</div>
`;

exports[`InfiniteHits - Raw button is disabled when it is the last page 1`] = `
<div>
  <ul
    className="list"
  >
    <li
      className="item"
      key="0"
    >
      <div
        style={
          Object {
            "borderBottom": "1px solid #bbb",
            "marginBottom": "5px",
            "paddingBottom": "5px",
            "wordBreak": "break-all",
          }
        }
      >
        {"objectID":0}
        ...
      </div>
    </li>
    <li
      className="item"
      key="1"
    >
      <div
        style={
          Object {
            "borderBottom": "1px solid #bbb",
            "marginBottom": "5px",
            "paddingBottom": "5px",
            "wordBreak": "break-all",
          }
        }
      >
        {"objectID":1}
        ...
      </div>
    </li>
    <li
      className="item"
      key="2"
    >
      <div
        style={
          Object {
            "borderBottom": "1px solid #bbb",
            "marginBottom": "5px",
            "paddingBottom": "5px",
            "wordBreak": "break-all",
          }
        }
      >
        {"objectID":2}
        ...
      </div>
    </li>
  </ul>
  <button
    className="loadMore loadMore--disabled"
    disabled={true}
    onClick={[Function]}
  >
    loadMore
  </button>
</div>
`;

exports[`InfiniteHits - Raw renders a custom hit 1`] = `
<div>
  <ul
    className="list"
  >
    <li
      className="item"
      key="0"
    >
      <div
        id={0}
      />
    </li>
    <li
      className="item"
      key="1"
    >
      <div
        id={1}
      />
    </li>
    <li
      className="item"
      key="2"
    >
      <div
        id={2}
      />
    </li>
  </ul>
  <button
    className="loadMore loadMore--disabled"
    disabled={true}
    onClick={[Function]}
  >
    loadMore
  </button>
</div>
`;

exports[`InfiniteHits - Raw renders a default hit 1`] = `
<div>
  <ul
    className="list"
  >
    <li
      className="item"
      key="0"
    >
      <div
        style={
          Object {
            "borderBottom": "1px solid #bbb",
            "marginBottom": "5px",
            "paddingBottom": "5px",
            "wordBreak": "break-all",
          }
        }
      >
        {"objectID":0}
        ...
      </div>
    </li>
    <li
      className="item"
      key="1"
    >
      <div
        style={
          Object {
            "borderBottom": "1px solid #bbb",
            "marginBottom": "5px",
            "paddingBottom": "5px",
            "wordBreak": "break-all",
          }
        }
      >
        {"objectID":1}
        ...
      </div>
    </li>
    <li
      className="item"
      key="2"
    >
      <div
        style={
          Object {
            "borderBottom": "1px solid #bbb",
            "marginBottom": "5px",
            "paddingBottom": "5px",
            "wordBreak": "break-all",
          }
        }
      >
        {"objectID":2}
        ...
      </div>
    </li>
  </ul>
  <button
    className="loadMore loadMore--disabled"
    disabled={true}
    onClick={[Function]}
  >
    loadMore
  </button>
</div>
`;
